docs/content/library/AsyncArrow.fsx

```diff
@@ -0,0 +1,339 @@
+reified type for implementing service pipelines.
+
+The `AsyncArrow` type is located in the `FSharpx.Async.dll` assembly which can be loaded in F# Interactive as follows:
+*)
+
+#r "../../../bin/FSharpx.Async.dll"
+#r "System.Net.Http.dll"
+open FSharpx.Control
+
+
+(**
+## Motivation
+
+For example, consider an HTTP request-reply protocol as provided
+by [System.Net.Http.HttpClient](https://msdn.microsoft.com/en-us/library/system.net.http.httpclient%28v=vs.118%29.aspx). 
+The method [HttpClient.SendAsync](https://msdn.microsoft.com/en-us/library/hh138176(v=vs.118).aspx) 
+when adapted to `Async` has type `HttpRequestMessage -> Async<HttpResponseMessage>`. A specific instance
+of `HttpClient` can be constructed and then passed around as a function value. For brevity we also alias
+the HTTP message types:
+*)
+
+open System.Net
+open System.Net.Http
+
+type HttpReq = HttpRequestMessage
+
+type HttpRes = HttpResponseMessage
+
+let httpClient = new HttpClient()
+
+let httpService : HttpReq -> Async<HttpRes> = 
+  httpClient.SendAsync >> Async.AwaitTask
+
+
+(**
+Suppose you wanted to modify this function such that a header is added to the HTTP response before it is returned. This can be
+done as follows:
+*)
+
+let httpServiceWithHeader : HttpReq -> Async<HttpRes> =
+  httpService >> Async.map (fun res ->
+    res.Content.Headers.Add("X-Hello", "World")
+    res)
+
+(**
+Similarly to modify the headers of the incoming HTTP request:
+*)
+
+let httpServiceWithHeader' : HttpReq -> Async<HttpRes> =
+  fun req -> 
+    req.Headers.Add("X-Foo", "Bar")
+    httpServiceWithHeader req
+
+(**
+Now the value `httpServiceWithHeader'` is an HTTP service which calls `HttpClient` but adds a header to both
+the incoming HTTP request and to the outgoing HTTP response. Up to this point we've relied on existing mechanisms
+provided by `Async` itself. Suppose next that we wanted to measure the latency of HTTP interactions. This can be
+done as follows:
+*)
+
+let httpServiceTimed : HttpReq -> Async<HttpRes> =
+  fun req -> async {
+    let sw = System.Diagnostics.Stopwatch.StartNew()
+    let! res = httpServiceWithHeader' req
+    sw.Stop()
+    printfn "elapsed_ms=%i" sw.ElapsedMilliseconds
+    return res
+  }
+
+(**
+This timing logic can be useful outside of the specific HTTP service that we've composed. Let us abstract it:
+*)
+
+let httpTimer (service:HttpReq -> Async<HttpRes>) (req:HttpReq) =
+  async {
+    let sw = System.Diagnostics.Stopwatch.StartNew()
+    let! res = service req
+    sw.Stop()
+    printfn "elapsed_ms=%i" sw.ElapsedMilliseconds
+    return res    
+  }
+
+(**
+The value `httpTimer` is a function which takes an HTTP service and produces another HTTP service which wraps the
+argument service with a timer. If we view the HTTP service as an async arrow `AsyncArrow<HttpReq, HttpRes>`
+then the above value is a mapping between async arrows - an async filter of type `AsyncFilter<HttpReq, HttpRes>`.
```

Comment:
AsyncFilter is a strange name to use - it's not a filter as in "List.filter".  I don't even know if this concept needs a name - it's just a functional combinator like ``(T -> U) -> (T ->U)`` is a functional combinator.